### ISSUE TITLE
update hiera_xxx() to lookup() for hiera v5

### ISSUE THREAD

**Issue by 7yl4r** [voxpupuli/puppet-telegraf#65]
Using hiera 5 requires using the lookup function in place of hiera_xxx() functions. ([hiera lookup docs](https://docs.puppet.com/puppet/4.10/hiera_intro.html#some-features-are-deprecated-when-are-they-getting-removed)). 

Without updating this, deep merging using hiera 5 is impossible with this module, but updating will break compatibility with hiera 3.

**Comment by 7yl4r** [comment]
More specifically:
```diff
-  $_outputs = hiera_hash('telegraf::outputs', $outputs)
-  $_inputs = hiera_hash('telegraf::inputs', $inputs)

+  $_outputs = lookup({
+      name          => 'telegraf::outputs',
+      default_value => $outputs,
+  })
+  $_inputs = lookup({
+      name          => 'telegraf::inputs',
+      default_value => $inputs,
+   })
```

**Comment by jschaeff** [comment]
As a work around, you can configure hiera 5 for merging strategy on a specified parameter:
In your common.yaml for instance:

```yaml
lookup_options:
  telegraf::inputs:
    merge:
      strategy: deep
      merge_hash_arrays: true
```

**Comment by jschaeff** [comment]
You can get this behaviour by adding the variable in your lookup_options; 

```yaml
--- # common.yaml
lookup_options:
  telegraf::inputs:
    merge:
      strategy: deep
      merge_hash_arrays: true

**Comment by trunet** [comment]
> You can get this behaviour by adding the variable in your lookup_options;

hiera_hash is deprecated... lookup function needs to be used, doesn't matter the merge strategy

**Comment by yankcrime** [comment]
Fixed via #82.
